src/targets.rs

```diff
@@ -241,70 +396,132 @@ pub fn default_binary_format(triple: &Triple) -> BinaryFormat {
-            Architecture::Riscv32imc => "riscv32imc",
-            Architecture::Riscv64 => "riscv64",
-            Architecture::S390x => "s390x",
-            Architecture::Sparc => "sparc",
-            Architecture::Sparc64 => "sparc64",
-            Architecture::Sparcv9 => "sparcv9",
-            Architecture::Thumbv6m => "thumbv6m",
-            Architecture::Thumbv7a => "thumbv7a",
-            Architecture::Thumbv7em => "thumbv7em",
-            Architecture::Thumbv7m => "thumbv7m",
-            Architecture::Thumbv7neon => "thumbv7neon",
-            Architecture::Thumbv8mBase => "thumbv8m.base",
-            Architecture::Thumbv8mMain => "thumbv8m.main",
-            Architecture::Wasm32 => "wasm32",
-            Architecture::X86_64 => "x86_64",
+            ArmArchitecture::Arm => "arm",
+            ArmArchitecture::Aarch64 => "aarch64",
+            ArmArchitecture::Armv4 => "armv4",
+            ArmArchitecture::Armv4t => "armv4t",
+            ArmArchitecture::Armv5t => "armv5t",
+            ArmArchitecture::Armv5te => "armv5te",
+            ArmArchitecture::Armv5tej => "armv5tej",
+            ArmArchitecture::Armv6 => "armv6",
+            ArmArchitecture::Armv6j => "armv6j",
+            ArmArchitecture::Armv6k => "armv6k",
+            ArmArchitecture::Armv6z => "armv6z",
+            ArmArchitecture::Armv6kz => "armv6kz",
+            ArmArchitecture::Armv6t2 => "armv6t2",
+            ArmArchitecture::Armv6m => "armv6m",
+            ArmArchitecture::Armv7 => "armv7",
+            ArmArchitecture::Armv7a => "armv7a",
+            ArmArchitecture::Armv7ve => "armv7ve",
+            ArmArchitecture::Armv7m => "armv7m",
+            ArmArchitecture::Armv7r => "armv7r",
+            ArmArchitecture::Armv7s => "armv7s",
+            ArmArchitecture::Armv8 => "armv8",
+            ArmArchitecture::Armv8a => "armv8a",
+            ArmArchitecture::Armv8_1a => "armv8.1a",
+            ArmArchitecture::Armv8_2a => "armv8.2a",
+            ArmArchitecture::Armv8_3a => "armv8.3a",
+            ArmArchitecture::Armv8_4a => "armv8.4a",
+            ArmArchitecture::Armv8_5a => "armv8.5a",
+            ArmArchitecture::Armv8mBase => "armv8m.base",
+            ArmArchitecture::Armv8mMain => "armv8m.main",
+            ArmArchitecture::Armv8r => "armv8r",
+            ArmArchitecture::Thumbv6m => "thumbv6m",
+            ArmArchitecture::Thumbv7a => "thumbv7a",
+            ArmArchitecture::Thumbv7em => "thumbv7em",
+            ArmArchitecture::Thumbv7m => "thumbv7m",
+            ArmArchitecture::Thumbv7neon => "thumbv7neon",
+            ArmArchitecture::Thumbv8mBase => "thumbv8m.base",
+            ArmArchitecture::Thumbv8mMain => "thumbv8m.main",
+            ArmArchitecture::Armebv7r => "armebv7r",
         };
         f.write_str(s)
     }
 }
 
+impl fmt::Display for Architecture {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Architecture::Arm(arm) = *self {
```

Comment:
Since I expect we'll have similar enums for other architectures in the future, I'd prefer to start this as a `match` rather than an `if let`.